airflow/www/security.py

```diff
@@ -243,11 +243,9 @@ def __init__(self, appbuilder) -> None:
 
     def _get_root_dag_id(self, dag_id: str) -> str:
         if "." in dag_id:
-            dm = (
-                self.appbuilder.get_session.query(DagModel.dag_id, DagModel.root_dag_id)
-                .filter(DagModel.dag_id == dag_id)
-                .first()
-            )
+            dm = self.appbuilder.get_session.execute(
+                select(DagModel.dag_id, DagModel.root_dag_id).where(DagModel.dag_id == dag_id)
+            ).one()
```

Comment:
Since `first` was used originally, we should use `limit(1)` because using `one()` will throw error

Reply:
I think `limit(1)` probably doesn’t matter since DagModel has a primary key on dag_id. If this query returns more than one result we’re in big trouble.

Reply:
yeah, hence didnt make any change